pghoard/monitoring/prometheus.py

```diff
@@ -0,0 +1,47 @@
+"""
+Prometheus client (used to create a Prometheus endpoint)
+
+"""
+
+
+class PrometheusClient:
+    def __init__(self, config):
+        self._tags = config.get("tags", {})
+
+    def gauge(self, metric, value, tags=None):
+        self._update(metric, value, tags)
+
+    def increase(self, metric, inc_value=1, tags=None):
+        self._update(metric, inc_value, tags)
+
+    def timing(self, metric, value, tags=None):
+        self._update(metric, value, tags)
+
+    def unexpected_exception(self, ex, where, tags=None):
+        pass
+
+    def _update(self, metric, value, tags):
+        metric = metric.replace(".", "_")
+        tags = {**self._tags, **tags}
+        tag_list = []
+        for k in sorted(tags.keys()):
+            tag_list.append("{}=\"{}\"".format(k, tags[k]))
+        encoded_tags = "{{{}}}".format(",".join(tag_list))
+        formatted_metric = "{}{}".format(metric, encoded_tags)
+
+        f = open("/tmp/pghoard-prometheus.txt", "a+")
```

Comment:
I dislike this writing into a file/reading from a file. How about instead you write these to a local dictionary in PrometheusClient() and  just pass the self.metrics object to the webserver object and have it read the data from the dict and return it over HTTP?